### PR TITLE
Bulk Quick Add: Clicking "-" takes you back to 0

### PR DESCRIPTION
**Pull request by sofiamatulis**
### PR Summary: 

Clicking on `-` takes you back to 0 on live cart. This shouldn't affect regular add to cart

### Why are these changes introduced?

Fixes [#issue.](https://github.com/orgs/Shopify/projects/6397/views/26?pane=issue&itemId=52402255)

### Visual impact on existing themes
The buyer will be able to remove items from cart in live cart by using `-`

### Testing steps/scenarios

- [ ] Test cart (with and without vol pricing, qty rules) updating items in cart
- [ ] Test cart drawer updating items in cart (with and without vol pricing, qty rules) 
- [ ] Test adding to cart from PDP, feat product (regular add to cart - this shouldnt be affected)
- [ ] Test Quick Order List (with and without vol pricing, qty rules) updating items in cart
- [ ] Test Quick Add bulk in the collection page (with and without vol pricing, qty rules) updating items in cart

### Demo links

- [Editor](https://admin.shopify.com/store/os2-demo/themes/162952740886/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by scottgmeadows** [comment]
Something is broken in this PR adding to cart when there are quantity rules. Wasn't able to test - taking you back to 0.

[Here's the problem I experienced.](https://screenshot.click/27-14-8x5ax-pl52a.mp4)

Repro steps:
1) View a product with volume pricing and quantity rules as a b2b customer (eg: something with increments of 5, and a min of 10)
2) Click plus button
3) You'll get an error that says product has a min of 10 - but in dawn production we automatically infer the first valid value. 
4) Also a something is wrong error at the bottom.

**Comment by sofiamatulis** [comment]
> Something is broken in this PR adding to cart when there are quantity rules. Wasn't able to test - taking you back to 0.
> 
> [Here's the problem I experienced.](https://screenshot.click/27-14-8x5ax-pl52a.mp4)
> 
> Repro steps:
> 
> 1. View a product with volume pricing and quantity rules as a b2b customer (eg: something with increments of 5, and a min of 10)
> 2. Click plus button
> 3. You'll get an error that says product has a min of 10 - but in dawn production we automatically infer the first valid value.
> 4. Also a something is wrong error at the bottom.

@scottgmeadows Good catch 👍 this was happening when we had a stepcount lower than the min. This should be fixed now

**Comment by eugenekasimov** [comment]
I did the first pass and everything looks good except that the PDP was a bit effected in my opinion. I'm going to test more. 
27-53-5w4rg-8x9uh

File: snippets/quantity-input.liquid [hunk 1]
```diff
@@ -24,7 +24,8 @@
     {% # theme-check-disable %}
     value="{{ cart | item_count_for_variant: variant.id }}"
     data-cart-quantity="{{ cart | item_count_for_variant: variant.id }}"
-    min="{{ variant.quantity_rule.min }}"
+    min="0"
```
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply snippets/quantity-input.liquid]: Yeah, you are right. 

Thanks for fixing it. Unfortunately, there is still a minor issue with PDP quantity picker. Although now the "-" button is disabled I still can click on it. 

27-03-5rvxw-iuse3

**Review by eugenekasimov — APPROVED** [review]
Works as expected! 🚀

**Comment by sofiamatulis** [comment]
> I did some more testing and noticed that when **Enable quick add button** is checked within the featured collection section on the homepage or PDP, it doesn't work ([video](https://screenshot.click/29-16-rk6b3-0mx1d-rk6b3-0mx1d.mp4)). I also checked the editor link from #3217 and the same thing occurs. I'm assuming this is intentional and that it'll be taken care of in another PR? 🤔

Yes! This is not part of this functionality of this PR and will be taken care of when we introduce the new quick add to the remaining portions of the theme

**Review by scottgmeadows — APPROVED** [review]
Tested all the scenarios and LGTM :)

**Pull request merged**